src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router";
+import ArticleBlock from "./ArticleBlock";
+import CommentsBlock from "./CommentsBlock";
+
+function SingleArticle({ articleId }) {
+  const navigate = useNavigate();
+  const [oneArticle, setOneArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    articleId > 0 &&
+      axios
+        .get(`https://nc-news-api-qfui.onrender.com/api/articles/${articleId}`)
+        .then(({ data }) => {
+          setOneArticle(data.article);
+          setIsLoading(false);
+        });
+
+    articleId > 0 &&
+      axios
+        .get(
+          `https://nc-news-api-qfui.onrender.com/api/articles/${articleId}/comments`
+        )
+        .then(({ data }) => {
+          setComments(data.comments);
+          setIsLoading(false);
+        });
```

Comment:
I have a few comments for the code in this `useEffect`

- you're checking `articleId > 0` but that seems unnecessary because the Axios call should fail if the ID is less than 0, you could clean the code up by removing those conditions
- you are setting the same `isLoading` to be false in both promise chains, but one request may conclude before the other, or one may not conclude at all 
- By running these calls side by side there is a risk that the comments would load before the article, it might be wise to fetch the article first and then chain the comment fetching API call to happen afterwards 💡